scripts/ConcordanceMakeSamplesheet.sh

```diff
@@ -0,0 +1,217 @@
+                        ;;
+        esac
+done
+
+#
+# Check commandline options.
+#
+if [[ -z "${NGSGROUP:-}" ]]
+then
+        log4Bash 'FATAL' "${LINENO}" "${FUNCNAME:-main}" '1' 'Must specify an ngs-group with -g. For the ngs.vcf files'
+fi
+
+if [[ -z "${ARRAYGROUP:-}" ]]
+then
+        log4Bash 'FATAL' "${LINENO}" "${FUNCNAME:-main}" '1' 'Must specify an array-group with -a. for the array.vcf files'
+fi
+#
+# Source config files.
+#
+log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Sourcing config files..."
+declare -a configFiles=(
+        "${CFG_DIR}/${NGSGROUP}.cfg"
+        "${CFG_DIR}/${HOSTNAME_SHORT}.cfg"
+        "${CFG_DIR}/sharedConfig.cfg"
+        "${HOME}/molgenis.cfg"
+)
+
+for configFile in "${configFiles[@]}"; do 
+        if [[ -f "${configFile}" && -r "${configFile}" ]]
+        then
+                log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Sourcing config file ${configFile}..."
+                #
+                # In some Bash versions the source command does not work properly with process substitution.
+                # Therefore we source a first time with process substitution for proper error handling
+                # and a second time without just to make sure we can use the content from the sourced files.
+                #
+                mixed_stdouterr=$(source ${configFile} 2>&1) || log4Bash 'FATAL' ${LINENO} "${FUNCNAME:-main}" ${?} "Cannot source ${configFile}."
+                source ${configFile}  # May seem redundant, but is a mandatory workaround for some Bash versions.
+        else
+                log4Bash 'FATAL' "${LINENO}" "${FUNCNAME:-main}" '1' "Config file ${configFile} missing or not accessible."
+        fi
+done
+
+#
+# Make sure to use an account for cron jobs and *without* write access to prm storage.
+#
+
+if [[ "${ROLE_USER}" != "${ATEAMBOTUSER}" ]]
+then
+        log4Bash 'FATAL' "${LINENO}" "${FUNCNAME:-main}" '1' "This script must be executed by user ${ATEAMBOTUSER}, but you are ${ROLE_USER} (${REAL_USER})."
+fi
+
+#
+##       This script will match the ngs.vcf with the array.vcf, and will make a small sample sheet.
+###      This samplesheet is the input for the actual concordance check.
+##      
+# 
+
+module load HTSlib
+module load BEDTools
+module list
+
+ngsVcfDir="/groups/${NGSGROUP}/${TMP_LFS}/Concordance/ngs/"
+concordanceDir="/groups/${NGSGROUP}/${TMP_LFS}/Concordance/"
+arrayVcfDir="/groups/${ARRAYGROUP}/${TMP_LFS}/Concordance/array/"
+
+for vcfFile in $(ls "${ngsVcfDir}"*"final.vcf")
+do
+    echo "_____________________________________________________________" ##Can be removed later, more easy to see when a new sample is processed
+    log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "processing ngs-vcf ${vcfFile}"
+    ngsVcfId=$(basename "${vcfFile}" .final.vcf)
+    ngsBarcode=$(grep "##FastQ_Barcode=" "${vcfFile}" | awk 'BEGIN {FS="="}{OFS="_"} {print _,$2}')
```

Comment:
if no FastQ_Barcode exists this script will error

Reply:
ik heb dit getest, en ngsBarcode is dan leeg. Script maak wel gewoon een samplesheet, die werkt voor de concordance check.

Reply:
grappig dat dit dan wel werkt, als je alleen een grep commando doet zonder het door te pipen naar awk dan werkt het niet en stopt het script.. Dit werkt inderdaad wel